### PR TITLE
Symfony 5.4 deprecations adressed

### PR DESCRIPTION
**Pull request by Eloar**
Noticed some deprecation messages from RabbitMqBundle when upgraded my project from Symfony 5.3 to Symfony 5.4. Added some more type hinting in Extension class.

### CODE REVIEW

**Comment by mihaileu** [comment]
@Eloar  is it working with older sy versions ?

**Comment by Eloar** [comment]
@mihaileu based on tests/checks it does work with lowest version currently supported (Symfony 4.3).

**Comment by mihaileu** [comment]
I think we can merge it after merge this one: https://github.com/php-amqplib/RabbitMqBundle/pull/675 as we will lose that older sy compatibility

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ public function __construct($name)
         $this->name = $name;
     }
 
-    public function getConfigTreeBuilder()
+    public function getConfigTreeBuilder() : TreeBuilder
```
Comment (mihaileu) [inline DependencyInjection/Configuration.php:32]: if there is a space between ) and : we should eliminate it and stick to the PSR12

Comment (Eloar) [reply DependencyInjection/Configuration.php]: thx for pointing that out. changed all the places I made that mistake and updated commit. It is now ready to be merged if no other issues appear.

File: phpunit.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (javer) [inline phpunit.xml:1]: Why is this file committed when there is the identical one in https://github.com/php-amqplib/RabbitMqBundle/blob/master/phpunit.xml.dist ?

File: DependencyInjection/OldSoundRabbitMqExtension.php [hunk 3]
```diff
@@ -71,7 +72,7 @@ public function load(array $configs, ContainerBuilder $container)
         }
     }
 
-    public function getConfiguration(array $config, ContainerBuilder $container)
+    public function getConfiguration(array $config, ContainerBuilder $container): ?ConfigurationInterface
```
Comment (javer) [inline DependencyInjection/OldSoundRabbitMqExtension.php:75]: It's unlikely that this method will return null ever.

**Review by javer — CHANGES_REQUESTED** [review]

File: DependencyInjection/OldSoundRabbitMqExtension.php (hunk 3 above)
Comment (mihaileu) [reply DependencyInjection/OldSoundRabbitMqExtension.php:75]: deleted

**Review by mihaileu — APPROVED** [review]

**Pull request merged**